hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -146,7 +154,7 @@ class MergeOnReadSnapshotRelation(val sqlContext: SQLContext,
     rdd.asInstanceOf[RDD[Row]]
   }
 
-  def buildFileIndex(filters: Array[Filter]): List[HoodieMergeOnReadFileSplit] = {
+  def buildFileIndex(filters: Array[Filter]): List[List[HoodieMergeOnReadFileSplit]] = {
```

Comment:
> may I know why this change?

For Spark, it can treat the hudi table with bucket index as a bucketed table and then each task will read the files  belonging to the same bucket. If reading only one partition, a task will only read a file group. But if more than one, to bucket table, each task will read the files which has the same bucket id and from different partition. Thus each task  will have more one file split.
Treating as a bucketed table will accelerated query, especially to the join/group by bucket index key query.